gateway/api/services/result_storage.py

```diff
@@ -0,0 +1,78 @@
+"""
+This module handle the access to the result store
+"""
+import os
+import logging
+from typing import Optional
+from django.conf import settings
+
+logger = logging.getLogger("gateway")
+
+
+class ResultStorage:
+    """Handles the storage and retrieval of user job results."""
+
+    RESULT_FILE_EXTENSION = ".json"
+    ENCODING = "utf-8"
+
+    def __init__(self, username: str):
+        """Initialize the storage path for a given user."""
+        self.user_results_directory = os.path.join(
+            settings.MEDIA_ROOT, username, "results"
+        )
+        os.makedirs(self.user_results_directory, exist_ok=True)
+
+    def __get_result_path(self, job_id: str) -> str:
+        """Construct the full path for a result file."""
+        return os.path.join(
+            self.user_results_directory, f"{job_id}{self.RESULT_FILE_EXTENSION}"
+        )
+
+    def get(self, job_id: str) -> Optional[str]:
+        """
+        Retrieve a result file for the given job ID.
+
+        Returns:
+            Tuple containing:
+            - FileWrapper for the file
+            - File MIME type
+            - File size in bytes
+        """
+        result_path = self.__get_result_path(job_id)
+        if not os.path.exists(result_path):
+            logger.warning(
```

Comment:
How we are assuming that some jobs are not going to have this file let's use `logger.info` here.